core/src/main/java/org/mskcc/cbio/portal/dao/DaoMutation.java

```diff
@@ -137,10 +137,22 @@ public static void createMutationCountClinicalData(GeneticProfile geneticProfile
         ResultSet rs = null;
         try {
             con = JdbcUtil.getDbConnection(DaoMutation.class);
+            // add mutation count meta attribute if it does not exist
+            ClinicalAttribute clinicalAttribute = DaoClinicalAttributeMeta.getDatum(MUTATION_COUNT_ATTR_ID, geneticProfile.getCancerStudyId());
+            if (clinicalAttribute == null) {
+                ClinicalAttribute attr = new ClinicalAttribute(MUTATION_COUNT_ATTR_ID, "Mutation Count", "Mutation Count", "NUMBER",
+                    false, "30", geneticProfile.getCancerStudyId());
+                DaoClinicalAttributeMeta.addDatum(attr);
+            }
+
             // do not include germline and fusions (msk internal) when counting mutations
             // we do not add the MUTATION_COUNT clinical data for the sample if it's not profiled
+            // use replace to overwrite existing counts if they exist (e.g. when
+            // the mutation data is split over multiple files with same profile
+            // id)
             pstmt = con.prepareStatement(
-                    "SELECT sample_profile.`SAMPLE_ID`, COUNT(DISTINCT mutation_event.`CHR`, mutation_event.`START_POSITION`, " +
+                    "REPLACE `clinical_sample` " +
```

Comment:
REPLACE is an extension to standard SQL syntax ... we are already tied to the mysql DBMS, so maybe this is not an issue, but this would be one more thing that a deployer using a different (DB2?) DBMS would need to port.
Also, I needed to look up this syntax and read docs in order to know what was being done, so there is additional overhead on developers.
On the other hand, this command does seem to do exactly what we need, and concisely.

Reply:
The differing dbms argument will become less of an issue as we move to container based deployments. So not too worried about this in the long term.

I can add a comment in the code that replace is basically insert+update depending on whether data is available